src/core/api/public_api.ts

```diff
@@ -711,50 +721,100 @@ class Player extends EventEmitter<IPublicAPIEvent> {
       throw new Error("the attached video element is disposed");
     }
 
-    // now that every checks have passed, stop previous content
-    this.stop();
-
     const isDirectFile = transport === "directfile";
 
-    this._priv_currentError = null;
-    this._priv_contentInfos = { url,
-                                isDirectFile,
-                                segmentBuffersStore: null,
-                                thumbnails: null,
-                                manifest: null,
-                                currentPeriod: null,
-                                activeAdaptations: null,
-                                activeRepresentations: null,
-                                initialAudioTrack: defaultAudioTrack,
-                                initialTextTrack: defaultTextTrack };
-
-    // inilialize `_priv_playing$` to false (meaning the content is not playing yet)
-    this._priv_playing$.next(false);
+    /** Subject which will emit to stop the current content. */
+    const stopContent$ = new Subject<void>();
+
+    /** Future `this._priv_contentInfos` related to this content. */
+    const contentInfos = { url,
+                           stop$: stopContent$,
+                           isDirectFile,
+                           segmentBuffersStore: null,
+                           thumbnails: null,
+                           manifest: null,
+                           currentPeriod: null,
+                           activeAdaptations: null,
+                           activeRepresentations: null,
+                           initialAudioTrack: defaultAudioTrack,
+                           initialTextTrack: defaultTextTrack };
+
+    /**
+     * Callback to call once we're ready to officially stop the previous content
+     * and consider the new one in our API.
+     * We don't want to do that right now because the `stop` operation might
+     * take time (from milliseconds to several hundred of milliseconds on some
+     * devices).
+     * Here we can optimize that time by for example doing that while waiting on
+     * a HTTP request to respond.
+     */
+    const resetState = () => {
+      this.stop();
+      this._priv_currentError = null;
```

Comment:
And why not move the  `this._priv_currentError = null;` and `this._priv_playing$.next(false);` in the stop function ?
I mean, we call the stop function twice, here, and in the dispose function. It would not be a problem to unset the current error and emitting a last time on the playing subject, am I wrong?

Reply:
(The aim to get rid of the reset state function)

Reply:
I'm not sure, as `stop` is more importantly an API.
Perhaps it would still make a sense there but I'm not too sure. To think about.

Reply:
I removed the `resetState` function like we discussed.
Instead I copied its content everywhere it was called with an added comment